Sources/Metrics/Metrics.swift

```diff
@@ -24,6 +24,9 @@ extension Timer {
     ///     - label: The label for the Timer.
     ///     - dimensions: The dimensions for the Timer.
     ///     - body: Closure to run & record.
+    #if compiler(>=6.0)
```

Comment:
Do we think it's okay to have a deprecation based on a compiler version? What's the experience for an adopter who supports the same 3 Swift versions as us, do they have to also provide two solutions?

Also, the new API has an OS requirement, so we should make sure we don't put any adopters into a situation where they can't fix this warning without bumping their OS requirement (which today requires a major bump).